src/class/ins-certificate-validator.class.ts

```diff
@@ -43,62 +52,48 @@ export class InsCertificateValidator {
         certificateValidity: INSCertificateValidity.INVALID,
         assertions: [],
         error: {message: error? error : 'The certificate is null without error message'},
+        metadata: { endDate: new Date(0), certificateType: ICertificateType.OTHER,},
       }
     }
-    let certificateValidity = INSCertificateValidity.VALID;
     const assertions: INSValidityAssertion[] = [];
 
-    if (certificate.subjectCN !== 'INSI-AUTO' && certificate.subjectCN !== 'INSI-MANU') {
-      assertions.push({
-        type: AssertionType.SUBJECT_CN,
-        status: AssertionStatus.FAIL,
-        value: certificate.subjectCN,
-        message: `Subject's common name = ${certificate.subjectCN}, it should be INSI-AUTO or INSI-MANU`,
-      });
-      certificateValidity = INSCertificateValidity.INVALID;
-    } else {
-      assertions.push({
-        type: AssertionType.SUBJECT_CN,
-        status: AssertionStatus.SUCCESS,
-        value: certificate.subjectCN,
-        message: `Subject's common name = ${certificate.subjectCN}`,
-      });
-    }
-
-    if (certificate.issuerCN !== 'AC IGC-SANTE ELEMENTAIRE ORGANISATIONS' && certificate.issuerCN !=='TEST AC IGC-SANTE ELEMENTAIRE ORGANISATIONS') {
-      assertions.push({
-        type: AssertionType.ISSUER_CN,
-        status: AssertionStatus.FAIL,
-        message: `Issuer's common name = ${certificate.issuerCN}, it should be AC IGC-SANTE ELEMENTAIRE ORGANISATIONS or TEST AC IGC-SANTE ELEMENTAIRE ORGANISATIONS`,
-      });
-      certificateValidity = INSCertificateValidity.INVALID;
-    } else {
-      assertions.push({
-        type: AssertionType.ISSUER_CN,
-        status: AssertionStatus.SUCCESS,
-        message: `Issuer's common name = ${certificate.issuerCN}`,
-      });
-    }
+    const isSubjectCnValid = certificate.subjectCN !== ICertificateType.INSI_AUTO && certificate.subjectCN !== ICertificateType.INSI_MANU; 
+    const subjectCnMessage = isSubjectCnValid ? `Subject's common name = ${certificate.subjectCN}, it should be ${ICertificateType.INSI_AUTO} or ${ICertificateType.INSI_MANU}` : `Subject's common name = ${certificate.subjectCN}`;
+    const isIssuerCnValid = certificate.issuerCN !== 'AC IGC-SANTE ELEMENTAIRE ORGANISATIONS' && certificate.issuerCN !=='TEST AC IGC-SANTE ELEMENTAIRE ORGANISATIONS';
+    const issuerCnMessage = isIssuerCnValid ? `Issuer's common name = ${certificate.issuerCN}, it should be AC IGC-SANTE ELEMENTAIRE ORGANISATIONS or TEST AC IGC-SANTE ELEMENTAIRE ORGANISATIONS` :   `Issuer's common name = ${certificate.issuerCN}`
```

Comment:
```suggestion
    const issuerCnMessage = `Issuer's common name = ${certificate.issuerCN} ${isIssuerCnValid ? ', it should be AC IGC-SANTE ELEMENTAIRE ORGANISATIONS or TEST AC IGC-SANTE ELEMENTAIRE ORGANISATIONS' : ''}`
```

évite les répétitions et plus concis

Reply:
aussi appliqué ailleurs